packages/parse5/lib/parser/index.ts

```diff
@@ -419,7 +420,7 @@ export class Parser<T extends TreeAdapterTypeMap> implements TokenHandler, Stack
         } else {
             const parent = this.openElements.currentTmplContentOrNode;
 
-            this.treeAdapter.appendChild(parent, element);
+            this.treeAdapter.appendChild(parent ?? this.document, element);
```

Comment:
This one looks a bit iffy to me, to append to the entire document? I would *guess* something like `if (parent) { this.treeAdapter(parent, element) }` makes more sense?
But I am missing the likely needed context, so if you think it’s :+1:, then I am :+1:

(this applies to both PRs I guess)

(otherwise :+1:)

Reply:
we could, its upto us what behaviour we want

basically, its possible to end up with a messy element stack in some rare cases, with broken syntax etc.

in those cases, the parent doesn't necessarily exist here. so we can either do nothing, or append to the root. if we do nothing, it may be confusing that nothing happened (empty tree). but I understand appending to the document may also be a surprise

I don't think the tests/spec define what to do here.

Reply:
@wooorm what do you think we should do about this?

Reply:
i'm not worried

i was just checking since i had written a response and not heard anything since